### PR TITLE
Druid simple jit

### PR DESCRIPTION
**Pull request by guillep**
Add support for simple druid JIT compiler.
- Add `SimpleDruidJIT`
- Add `PopN:`
- Do a pass on interpreter annotations

### CODE REVIEW

File: smalltalksrc/VMMaker/DruidJIT.class.st [hunk 1]
```diff
@@ -7282,8 +7166,7 @@ DruidJIT >> gen_StoreAndPopReceiverVariableBytecode0 [
 	jump3 jmpTarget: currentBlock.
 	jump6 jmpTarget: currentBlock.
 	jump1 jmpTarget: currentBlock.
-	self deoptimize.
-	deadCode := false.
+	self MoveR: t1 M64: 8 r: t0.
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/DruidJIT.class.st:7169]: ??? Is it right?

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 2]
```diff
@@ -12664,7 +12665,7 @@ StackInterpreter >> pushNewArrayBytecode [
 
 { #category : #'stack bytecodes' }
 StackInterpreter >> pushReceiverBytecode [
-	<needsFrameNever: 1>
+	"<needsFrameNever: 1>"
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:12668]: ???

**Review by PalumboN — APPROVED** [review]
I'm worry about the changes in `gen_StoreAndPopReceiverVariableBytecode`s.

The _StackToRegister DruidJIT_ is still working?

File: smalltalksrc/VMMaker/DruidJIT.class.st (hunk 1 above)
Comment (guillep) [reply smalltalksrc/VMMaker/DruidJIT.class.st:7169]: I think so, no? Why should we have two deoptimizes?

File: smalltalksrc/VMMaker/StackInterpreter.class.st (hunk 2 above)
Comment (guillep) [reply smalltalksrc/VMMaker/StackInterpreter.class.st:12668]: In the simple stack jit pushing the receiver sometimes needs a frame, sometimes not. So the annotation is not right for it.

I want to discuss this with you :)

File: smalltalksrc/VMMaker/DruidJIT.class.st (hunk 1 above)
Comment (PalumboN) [reply smalltalksrc/VMMaker/DruidJIT.class.st:7169]: Ahh yes! This is good... strange the old version...

**Pull request merged**